cmd/glasskube/cmd/list.go

```diff
@@ -23,72 +23,88 @@ type ListCmdOptions struct {
+		Namespace:     o.Namespace,
 	}
 }
 
 var listCmdOptions = ListCmdOptions{
 	KindOptions: DefaultKindOptions(),
 }
 
-var listCmd = &cobra.Command{
-	Use:     "list",
-	Aliases: []string{"ls", "l"},
-	Short:   "List packages",
-	Long: "List packages. By default, all available packages of the given repository are shown, " +
-		"as well as their installation status in your cluster.\nYou can choose to only show installed packages.",
-	PreRun: cliutils.SetupClientContext(true, &rootCmdOptions.SkipUpdateCheck),
-	Run: func(cmd *cobra.Command, args []string) {
-		ctx := cmd.Context()
-		if listCmdOptions.More {
-			listCmdOptions.ShowLatestVersion = true
-			listCmdOptions.ShowDescription = true
-			listCmdOptions.ShowMessage = true
-		}
-		lister := list.NewListerWithRepoCache(ctx)
-		var clPkgs []*list.PackageWithStatus
-		var pkgs []*list.PackagesWithStatus
-		var err error
-		if listCmdOptions.Kind != KindPackage {
-			clPkgs, err = lister.GetClusterPackagesWithStatus(ctx, listCmdOptions.toListOptions())
-			handleListErr(len(clPkgs), err, "clusterpackages")
-		}
-		if listCmdOptions.Kind != KindClusterPackage {
-			pkgs, err = lister.GetPackagesWithStatus(ctx, listCmdOptions.toListOptions())
-			handleListErr(len(pkgs), err, "packages")
-		}
-		noPkgs := len(pkgs) == 0 && listCmdOptions.Kind != KindClusterPackage
-		noClPkgs := len(clPkgs) == 0 && listCmdOptions.Kind != KindPackage
-		if noPkgs {
-			handleEmptyList("packages")
-		} else if len(pkgs) > 0 {
-			printPackageTable(pkgs)
-			if listCmdOptions.Kind != KindPackage {
-				fmt.Fprintln(os.Stderr, "")
+func newListCmd() *cobra.Command {
+	listCmd := &cobra.Command{
+		Use:     "list",
+		Aliases: []string{"ls", "l"},
+		Short:   "List packages",
+		Long: "List packages. By default, all available packages of the given repository are shown, " +
+			"as well as their installation status in your cluster.\nYou can choose to only show installed packages.",
+		PreRun: cliutils.SetupClientContext(true, &rootCmdOptions.SkipUpdateCheck),
+		Args:   cobra.MaximumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+			if listCmdOptions.More {
+				listCmdOptions.ShowLatestVersion = true
+				listCmdOptions.ShowDescription = true
+				listCmdOptions.ShowMessage = true
 			}
-		}
-		if noClPkgs {
-			handleEmptyList("clusterpackages")
-		} else if len(clPkgs) > 0 {
-			if listCmdOptions.Output == OutputFormatJSON {
-				printPackageJSON(clPkgs)
-			} else if listCmdOptions.Output == OutputFormatYAML {
-				printPackageYAML(clPkgs)
-			} else {
-				printClusterPackageTable(clPkgs)
+			if len(args) > 0 {
+				listCmdOptions.packageName = args[0]
+			}
+			if listCmdOptions.Kind == KindClusterPackage &&
+				(listCmdOptions.packageName != "" || listCmdOptions.Namespace != "") {
+				//TODO returning an error in RunE will also print cmd help; Desired or not?
+				return fmt.Errorf("argument or flag not supported with kind %s", KindClusterPackage)
```

Comment:
We prefer `Run` over `RunE` for this reason. Logging the error and calling `cliutils.ExitWithError()` instead should be fine.